### PR TITLE
Update Redis Module API; support for GCC 10 and Ubuntu/Hirsute (2)

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: src/version.h [hunk 1]
```diff
@@ -35,4 +36,26 @@
 
 #define REDISGEARS_MODULE_NAME "rg"
 
-#endif /* SRC_VERSION_H_ */
+typedef struct RedisVersion{
```
Comment (MeirShpilraien) [inline src/version.h:39]: Why this was moved here?

File: src/cluster.h [hunk 2]
```diff
@@ -36,5 +32,7 @@ int Cluster_RefreshCluster(RedisModuleCtx *ctx, RedisModuleString **argv, int ar
 int Cluster_ClusterSet(RedisModuleCtx *ctx, RedisModuleString **argv, int argc);
 int Cluster_ClusterSetFromShard(RedisModuleCtx *ctx, RedisModuleString **argv, int argc);
 
-
-#endif /* SRC_CLUSTER_H_ */
+#define VERIFY_CLUSTER_INITIALIZE(c) \
```
Comment (MeirShpilraien) [inline src/cluster.h:35]: Why this was moved here?

File: src/common.h [hunk 3]
```diff
@@ -5,16 +5,20 @@
  *      Author: root
  */
 
-#ifndef SRC_COMMON_H_
-#define SRC_COMMON_H_
+#pragma once
```
Comment (MeirShpilraien) [inline src/common.h:8]: Do we must use this?

File: src/common.h [hunk 4]
```diff
@@ -24,33 +28,10 @@
 
 extern Gears_dictType* dictTypeHeapIdsPtr;
 
-typedef struct RedisVersion{
```
Comment (MeirShpilraien) [inline src/common.h:27]: Why moving this?

File: src/global.c [hunk 5]
```diff
@@ -0,0 +1,6 @@
+
+#define REDISMODULE_MAIN
```
Comment (MeirShpilraien) [inline src/global.c:2]: Why do we need this file?

File: src/version.h (hunk 1 above)
Comment (rafie) [reply src/version.h:39]: It's version-related, seems logical. Do you prefer it like it was?

File: src/cluster.h (hunk 2 above)
Comment (rafie) [reply src/cluster.h:35]: It's cluster-related, seems logical. Do you prefer it like it was?

File: src/common.h (hunk 3 above)
Comment (rafie) [reply src/common.h:8]: It's more elegant (saves the bottom #endif). Any reason against using this?

File: src/global.c (hunk 5 above)
Comment (rafie) [reply src/global.c:2]: `global.c` is excluded in Makefile from the `-include $(SRCDIR)/common.h`, therefore allowing pre-defining `REDISMODULE_MAIN`.

File: src/version.h (hunk 1 above)
Comment (MeirShpilraien) [reply src/version.h:39]: I see this file more as holding constants of the RedisGears version. This is used as Redis version and potentially more commonly used, this is why I believe it should be on common.

File: src/cluster.h (hunk 2 above)
Comment (MeirShpilraien) [reply src/cluster.h:35]: I agree, we can leave it here.

File: src/common.h (hunk 3 above)
Comment (MeirShpilraien) [reply src/common.h:8]: Is it supported by all C compilers (or at least by the popular ones)?

File: src/redismodule.h [hunk 6]
```diff
@@ -447,267 +580,306 @@ typedef struct RedisModuleTypeMethods {
...
-#define RedisModule_IsAOFClient(id) ((id) == CLIENT_ID_AOF)
+#define RedisModule_IsAOFClient(id) ((id) == UINT64_MAX)
 
 // enterprise api
-int REDISMODULE_API_FUNC(RedisModule_AvoidReplicaTraffic)();
-int REDISMODULE_API_FUNC(RedisModule_ShardingGetKeySlot)(RedisModuleString *keyname);
-void REDISMODULE_API_FUNC(RedisModule_ShardingGetSlotRange)(int *first_slot, int *last_slot);
+REDISMODULE_API int (*RedisModule_AvoidReplicaTraffic)() REDISMODULE_ATTR;
+REDISMODULE_API int (*RedisModule_ShardingGetKeySlot)(RedisModuleString *keyname) REDISMODULE_ATTR;
+REDISMODULE_API void (*RedisModule_ShardingGetSlotRange)(int *first_slot, int *last_slot) REDISMODULE_ATTR;
```
Comment (MeirShpilraien) [inline src/redismodule.h:879]: You did not add those to REDISMODULE_GET_API

File: src/common.h (hunk 3 above)
Comment (rafie) [reply src/common.h:8]: gcc, clang, msc and intel (same as gcc) support it.

File: src/version.h (hunk 1 above)
Comment (rafie) [reply src/version.h:39]: no problem, will move it back.

**Review by MeirShpilraien — APPROVED** [review]

**Pull request merged**